### PR TITLE
add middleware pkg

### PR DESCRIPTION
**Pull request by veezhang**
- [x] StandardResponse

### CODE REVIEW

**Review by nianiaJR — APPROVED** [review]
LGTM

File: middleware/standard_response.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+				}
+				return h.params.GetErrCode(err)
+			}, func() *errorx.ErrCode {
+				return errorx.NewErrCode(errorx.CCInternalServer, 0, 0, "ErrInternalServer")
+			}), err)
+			e, _ = errorx.AsCodeError(err)
+		}
+		httpStatus = e.GetHTTPStatus()
+
+		if bodyType != StandardResponseBodyNone {
```
Comment (nianiaJR) [inline middleware/standard_response.go:99]: bodyType？why to distinguish it?

File: middleware/standard_response.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+		}
+	} else if n < len(bs) {
+		h.errorf("write response failed, actual bytes: %d, written bytes: %d", len(bs), n)
+	}
+}
+
+func (m *StandardResponse) Handler(next http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		next(w, r.WithContext(context.WithValue(r.Context(),
+			standardResponseCtxKey{},
```
Comment (nianiaJR) [inline middleware/standard_response.go:156]: standardResponseCtxKey no need to be uppercase？

File: middleware/standard_response.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/vesoft-inc/go-pkg/errorx"
+)
+
+const (
+	StandardResponseBodyJson StandardResponseBodyType = 0 // nolint:revive
+	StandardResponseBodyNone StandardResponseBodyType = 1
```
Comment (nianiaJR) [inline middleware/standard_response.go:14]: why not use string type？it's more meaningful

File: middleware/standard_response.go (hunk 1 above)
Comment (veezhang) [reply middleware/standard_response.go:99]: In some case, there donot need body.

File: middleware/standard_response.go (hunk 2 above)
Comment (veezhang) [reply middleware/standard_response.go:156]: This does not need to be exposed to avoid misuse.

**Pull request merged**